test/core.jl

```diff
@@ -3100,15 +3100,25 @@ Base.convert(::Type{Foo11874},x::Int) = float(x)
 
 # issue #9233
 let
-    err = @test_throws TypeError NTuple{Int, 1}
```

Comment:
okay fair enough, sounds like there's a good reason for the change. slightly breaking if any packages were doing this, but probably worth it